userauth_test.go

```diff
@@ -139,38 +165,195 @@ func TestUserTokenAuthenticator_ValidateSignature(t *testing.T) {
+	r.Header.Set("X-Amz-Security-Token", defaultToken[3:])
 	r.URL.Path = "/username/"
-	assert.Error(t, a.Authenticate(r))
+	brokenToken := a.Authenticate(r)
+	assert.Equal(t, "broken token (claims are empty): map[]", brokenToken.Error()[0:38])
 
 	r, _ = http.NewRequest("", "/", nil)
 	r.Host = "localhost"
-	r.Header.Set("X-Amz-Security-Token", "randomeyJraWQiOiJyc2ExIiwiYWxnIjoiUlMyNTYifQ.eyJzdWIiOiJjNTc3M2Y0MWQxN2QyN2JkNTNiMWU2Nzk0YWVkYzMyZDc5MDZlNzc5QGVsaXhpci1ldXJvcGUub3JnIiwiYXVkIjoiMTUxMzc2NDUtMzE1My00ZDQ5LTlkZGItNTk0MDI3Y2Q0Y2E3IiwiYXpwIjoiMTUxMzc2NDUtMzE1My00ZDQ5LTlkZGItNTk0MDI3Y2Q0Y2E3Iiwic2NvcGUiOiJnYTRnaF9wYXNzcG9ydF92MSBvcGVuaWQiLCJpc3MiOiJodHRwczpcL1wvbG9naW4uZWxpeGlyLWN6ZWNoLm9yZ1wvb2lkY1wvIiwiZXhwIjoxNTg1ODMxOTEwLCJpYXQiOjE1ODU4MjgzMTAsImp0aSI6IjdkMDU3NTVhLWRkNjktNDRjZS05MzIwLThlOGRiZjIwNzExNSJ9.bOsxAe8F7i78MfI1FBF2Wiy6g_uO74rOCFe3hBe0Qpf-RhswWnI0ys8EDfxJlqbPj39VX1n9gSphgd8ivGzsf0f00OvHqC17RSN0l6J3FUjyyNi2bWmaiejIzIfxXI0Kyyy45cr-NT5F5m95yuA1O-KpTGtRgDK5zDnhj3XIxBvIZX_pzhI9okY6FzL5fH7ZU4LhunP0iBJfgswK6l2Dy6A4fPhfVtZRU1EeJUBVTJ7YIY-FBvRHAGPgpLFvcOeN4WM8R6mwQ3hJcvcVPAjq1769meOkLsWxpPrbMvxfrtRjd4ANrnvwiEH-syV_ELXm2ntqMlFuFdoJ3CXY9pMzgw")
+	r.Header.Set("X-Amz-Security-Token", "random"+defaultToken)
 	r.URL.Path = "/username/"
 	assert.Error(t, a.Authenticate(r))
+	brokenToken2 := a.Authenticate(r)
+	assert.Equal(t, "broken token (claims are empty): map[]", brokenToken2.Error()[0:38])
+
+	// Delete the keys when testing is done or failed
+	defer os.RemoveAll(demoKeysPath)
+}
+
+func TestUserTokenAuthenticator_ValidateSignature_EC(t *testing.T) {
+	// Create temp demo rsa key pair
+	demoKeysPath := "demo-ec-keys"
+	demoPrKeyName := "/dummy.ega.nbis.se"
+	prKeyPath, pubKeyPath, err := MakeFolder(demoKeysPath)
+	if err != nil {
+		fmt.Printf("error on creating directory: %v", err)
+	}
+
+	err = CreateECkeys(prKeyPath, pubKeyPath)
+	if err != nil {
+		fmt.Printf("error on creating EC keys: %v", err)
+	}
+
+	var pubkeys map[string][]byte
+	jwtpubkeypath := demoKeysPath + "/public-key/"
+
+	a := NewValidateFromToken(pubkeys)
+	a.pubkeys = make(map[string][]byte)
+	a.getjwtkey(jwtpubkeypath)
+
+	// Parse demo private key
+	prKeyParsed, err := ParsePrivateECKey(prKeyPath, demoPrKeyName)
+	if err != nil {
+		fmt.Printf("error on parsing private key: %v", err)
+	}
```

Comment:
This is a test so you can use assert instead of printing the error as that won't be visible
```suggestion
      assert.NoError(t, err)
```

Reply:
Fixed